front/doc/component.md

```diff
@@ -0,0 +1,116 @@
+# `Component`
+
+--------------------------------------------------------------------------------
+
+The Component class is a custom HTML element that extends the HTMLElement class.
+It provides a foundation for creating reusable and customizable web components.
+
+This class is an abstract class, and must not be instantiated.
+
+--------------------------------------------------------------------------------
+
+# Redefined Methods
+
+## render
+
+Returns the HTML content to be rendered inside the component.
+
+```javascript
+render() {
+  const message = 'Hello World!'
+  return `
+    <div>
+      <h1>${message}</h1>
+    </div>
+    `;
+}
+```
+
+## style
+
+Returns the CSS content to be rendered inside the component.
```

Comment:
Returns the CSS content to be rendered inside the component. -> Returns the **specific** CSS content to be rendered inside the component.